flytekit/types/directory/types.py

```diff
@@ -358,7 +363,7 @@ def to_literal(
         if should_upload:
             if remote_directory is None:
                 remote_directory = ctx.file_access.get_random_remote_directory()
-            ctx.file_access.put_data(source_path, remote_directory, is_multipart=True)
+            ctx.file_access.put_data(source_path, remote_directory, is_multipart=True, batch_size=batch_size)
```

Comment:
do we have to worry about which version `batch_size` was introduced?

Reply:
sync'd offline and this parameter has been present for the past 2 years. Our [lower-bound constraint](https://github.com/flyteorg/flytekit/blob/master/setup.py#L46) in `setup.py` is more than enough.